modin/experimental/core/storage_formats/hdk/query_compiler.py

```diff
@@ -510,7 +510,11 @@ def _set_columns(self, columns):
         if self._modin_frame._has_unsupported_data:
             default_axis_setter(1)(self, columns)
         else:
-            self._modin_frame = self._modin_frame._set_columns(columns)
+            try:
+                self._modin_frame = self._modin_frame._set_columns(columns)
+            except NotImplementedError:
+                default_axis_setter(1)(self, columns)
+                self._modin_frame._has_unsupported_data = True
```

Comment:
shouldn't we also mark the frame as containing unsupported data when `.iloc`ing the same columns? (ex. `df.iloc[: ["col1", "col1"]]`)

Reply:
loc does not change the current frame, but creates a new one, with duplicate columns. I've added the assertion, as proposed below.